src/main/java/uk/co/ramp/statistics/StatisticsWriter.java

```diff
@@ -0,0 +1,70 @@
+package uk.co.ramp.statistics;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.springframework.stereotype.Service;
+import uk.co.ramp.io.csv.CsvWriter;
+import uk.co.ramp.io.types.StandardProperties;
+import uk.co.ramp.statistics.types.ImmutableRValueOutput;
+import uk.co.ramp.statistics.types.Infection;
+
+@Service
+public class StatisticsWriter {
+  private final StatisticsRecorder statisticsRecorder;
+  private final StandardProperties properties;
+
+  public StatisticsWriter(StatisticsRecorder statisticsRecorder, StandardProperties properties) {
+    this.statisticsRecorder = statisticsRecorder;
+    this.properties = properties;
+  }
+
+  public void output(int timeLimit, Writer statsWriter, Writer rValueWriter) throws IOException {
+    outputGeneralStats(statsWriter);
+    outputR(timeLimit, rValueWriter);
+  }
+
+  void outputR(int timeLimit, Writer rValueWriter) throws IOException {
+
+    Map<Integer, List<Infection>> inf = statisticsRecorder.getR0Progression();
+    MovingAverage movingAverage = new MovingAverage(7);
+    List<ImmutableRValueOutput> rValueOutputs = new ArrayList<>();
+    for (int i = 0; i < timeLimit; i++) {
+      List<Infection> orDefault = inf.get(i);
+      if (orDefault != null) {
+        int seeded = orDefault.stream().mapToInt(Infection::infections).sum();
+        double r = seeded / (double) orDefault.size();
+        movingAverage.add(r);
+
+        rValueOutputs.add(
+            ImmutableRValueOutput.builder()
+                .time(i)
+                .newInfectors(orDefault.size())
+                .newInfections(seeded)
+                .r(r)
+                .sevenDayAverageR(movingAverage.getAverage())
+                .build());
+      }
+    }
```

Comment:
I feel like this logic would sit better within the statisticsRecorder. We could have a new method for the rolling avg

Reply:
Moved, but will need interface change to expose it in statistics writer.